File: Source/Assemblies/Caching/Roblox.Caching/CacheabilitySettings.ts
import { ICacheabilitySettings } from './ICacheabilitySettings';

export class CacheabilitySettings implements ICacheabilitySettings {
	private _CollectionsAreCacheable: boolean = false;
	private _CountsAreCacheable: boolean = false;
	private _EntityIsCacheable: boolean = false;
	private _IDLookupsAreCacheable: boolean = false;
	private _IsNullCacheable: boolean = false;
	private _HasUnqualifiedCollections: boolean = false;
	private _IDLookupsAreCaseSensitive: boolean = false;

	public get CollectionsAreCacheable() {
		return this._CollectionsAreCacheable;
	}

	public get CountsAreCacheable() {
		return this._CountsAreCacheable;
	}

	public get EntityIsCacheable() {
		return this._EntityIsCacheable;
	}

	public get IDLookupsAreCacheable() {
		return this._IDLookupsAreCacheable;
	}

	public get IsNullCacheable() {
		return this._IsNullCacheable;
	}

	public get HasUnqualifiedCollections() {
		return this._HasUnqualifiedCollections;
	}

	public get IDLookupsAreCaseSensitive() {
		return this._IDLookupsAreCaseSensitive;
	}

	public constructor(
		collectionsAreCacheable: boolean = false,
		countsAreCacheable: boolean = false,
		entityIsCacheable: boolean = false,
		idLookupsAreCacheable: boolean = false,
		isNullCacheable: boolean = false,
		hasUnqualifiedCollections: boolean = false,
		idLookupsAreCaseSensitive: boolean = false,
	) {
		this._CollectionsAreCacheable = collectionsAreCacheable;
		this._CountsAreCacheable = countsAreCacheable;
		this._EntityIsCacheable = entityIsCacheable;
		this._IDLookupsAreCacheable = idLookupsAreCacheable;
		this._IsNullCacheable = isNullCacheable;
		this._HasUnqualifiedCollections = hasUnqualifiedCollections;
		this._IDLookupsAreCaseSensitive = idLookupsAreCaseSensitive;
	}
}

File: C#/CacheabilitySettings.cs
using System;

namespace Roblox.Caching {
	public class CacheabilitySettings : ICacheabilitySettings {
		private bool _CollectionsAreCacheable = default(bool);
		private bool _CountsAreCacheable = default(bool);
		private bool _EntityIsCacheable = default(bool);
		private bool _IDLookupsAreCacheable = default(bool);
		private bool _IsNullCacheable = default(bool);
		private bool _HasUnqualifiedCollections = default(bool);
		private bool _IDLookupsAreCaseSensitive = default(bool);

		internal bool CollectionsAreCacheable {
			get { return _CollectionsAreCacheable; }
		}

		internal bool CountsAreCacheable {
			get { return _CountsAreCacheable; }
		}

		internal bool EntityIsCacheable {
			get { return _EntityIsCacheable; }
		}

		internal bool IDLookupsAreCacheable {
			get { return _IDLookupsAreCacheable; }
		}

		internal bool IsNullCacheable {
			get { return _IsNullCacheable; }
		}
		
		internal bool HasUnqualifiedCollections {
			get { return _HasUnqualifiedCollections; }
		}

		internal bool IDLookupsAreCaseSensitive {
			get { return _IDLookupsAreCaseSensitive; }
		}

		public CacheabilitySettings(bool collectionsAreCacheable = false, bool countsAreCacheable = false, bool entityIsCacheable = false, bool idLookupsAreCacheable = false, bool isNullCacheable = false, bool hasUnqualifiedCollections = false, bool idLookupsAreCaseSensitive = false) {
			_CollectionsAreCacheable = collectionsAreCacheable;
			_CountsAreCacheable = countsAreCacheable;
			_EntityIsCacheable = entityIsCacheable;
			_IDLookupsAreCacheable = idLookupsAreCacheable;
			_IsNullCacheable = isNullCacheable
			_HasUnqualifiedCollections = hasUnqualifiedCollections;
			_IDLookupsAreCaseSensitive = idLookupsAreCaseSensitive;
		}
	}
}